### DISCUSSION TITLE
Conditional mapping by method parameter?

### DISCUSSION THREAD

**Discussion by Draggluon** [mapstruct/mapstruct#3086] [Q&A]
I have this jpa:
  

    public class CandidateRecommendationJpa extends AuditingEntityJpa {
      @Id
      @GeneratedValue(strategy = GenerationType.IDENTITY)
      @Column(name = "ID_CANDIDATO_RECOMENDACION", nullable = false)
      private Integer id;
    
      @ManyToOne
      @JoinColumn(name = "ID_CANDIDATO_EMISOR", nullable = false)
      private CandidateJpa candidateSender;
    
      @ManyToOne
      @JoinColumn(name = "ID_CANDIDATO_RECEPTOR", nullable = false)
      private CandidateJpa candidateReceiver;
    }

And I have to mapper with this entity:

    @Getter
    @Setter
    @AllArgsConstructor
    @NoArgsConstructor
    @Builder
    @EqualsAndHashCode
    public class CandidateRecommendation {
    
      private Integer id;   
    
      private Candidate candidate;

    }

Mapper:
    
    @Mapper(componentModel = "spring")
    public interface CandidateRecommendationJpaMapper {
    
      CandidateRecommendationJpaMapper INSTANCE = Mappers.getMapper(CandidateRecommendationJpaMapper.class);
    
      @Mapping(target = "candidate.id", source = "candidateSender.id")
      @Mapping(target = "candidate.name", source = "candidateSender.name")
      @Mapping(target = "candidate.login", source = "candidateSender.employee.login")
      CandidateRecommendation toModel(CandidateRecommendationJpa jpa);
    
      List<CandidateRecommendation> toModels(List<CandidateRecommendationJpa> jpa);
    
    }

**Problem**

I have to specify jpa atribute to map, is there a way to map a jpa attribute with a param method? For example, in this case I map with candidateSender but, how I can map with candidateReceiver?

      @Mapping(target = "candidate.id", source = "candidateReceiver.id")
      @Mapping(target = "candidate.name", source = "candidateReceiver.name")
      @Mapping(target = "candidate.login", source = "candidateReceiver.employee.login")
      CandidateRecommendation toSenderModel(CandidateRecommendationJpa jpa);
    
      List<CandidateRecommendation> toSenderModels(List<CandidateRecommendationJpa> jpa);

This show the following error:

    Ambiguous mapping methods found for mapping collection element to CandidateRecommendation: CandidateRecommendation toModel(CandidateRecommendationJpa jpa), CandidateRecommendation toSenderModel(CandidateRecommendationJpa jpa)

**Comment by chenzijia12300** [comment]
Hello ah, you can read this part of the [document](https://mapstruct.org/documentation/stable/reference/html/#selection-based-on-qualifiers).
For example, use the `qualifiedByName` property of the `IterableMapping` annotation with the `@Named` annotation to specify the conversion method
